apps/prairielearn/src/ee/lib/ai-grading.ts

```diff
@@ -44,34 +46,76 @@ const GradedExampleSchema = z.object({
   submission_text: z.string(),
   score_perc: z.number(),
   instance_question_id: z.string(),
+  manual_rubric_grading_id: z.string().nullable(),
 });
 type GradedExample = z.infer<typeof GradedExampleSchema>;
 
-async function generateGPTPromptWithExamples({
+const AppliedRubricItemSchema = z.object({
+  rubric_item_id: IdSchema,
+  score: z.coerce.number().nullish(),
+});
+type AppliedRubricItem = z.infer<typeof AppliedRubricItemSchema>;
+function parseRubricItems({
+  rubric_items,
+  gpt_rubric_items,
+}: {
+  rubric_items: RubricItem[];
+  gpt_rubric_items: number[];
+}): AppliedRubricItem[] {
+  const output: AppliedRubricItem[] = [];
+  for (const rubric_item_number of gpt_rubric_items) {
+    output.push(
+      AppliedRubricItemSchema.parse({ rubric_item_id: rubric_items[rubric_item_number].id }),
+    );
+  }
+  return output;
+}
+
+async function generateGPTPrompt({
   question_prompt,
   student_answer,
   example_submissions,
+  rubric_items,
```

Comment:
for variable names that live only in javascript, we usually use camelCase. We only use snake_case when a variable is going to or from SQL.
```suggestion
  questionPrompt,
  studentAnswer,
  exampleSubmissions,
  rubricItems,
```

Reply:
For this function, I'm using single parameter input with an object, and some of these inputs do come from a query. I wanted to avoid having something like `exampleSubmissions: example_submissions` in the function call so I'm using snake case for all. Should I change all of them to camel case, or only the ones not related to a query?

Reply:
Actually I just changed the variable names of these that do not come from a query.